synapse_data_warehouse/synapse/dynamic_tables/V2.22.0__accessrequirement_latest.sql

```diff
@@ -0,0 +1,22 @@
+use schema {{database_name}}.synapse; --noqa: JJ01,PRS,TMP,CP01
+
+CREATE DYNAMIC TABLE IF NOT EXISTS ACCESSREQUIREMENT_LATEST
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+    WITH latest_unique_rows AS (
+        SELECT
+            accessrequirementsnapshots.*,
+        FROM
+            {{database_name}}.synapse_raw.accessrequirementsnapshots --noqa: TMP
+        QUALIFY ROW_NUMBER() OVER (
+                PARTITION BY id
+                ORDER BY change_timestamp DESC,snapshot_timestamp DESC
```

Comment:
By relying solely on `snapshot_timestamp` to get the latest state of an item (in this case AR configuration), we run the risk of not getting the latest state **because `snapshot_timestamp` can also reflect the scheduled snapshots, not just the ones that are triggered from a change**.

Because of this, I'm also ordering the rows in the partitions by `change_timestamp` in addition to `snapshot_timestamp` as a precaution.

<img width="553" alt="image" src="https://github.com/user-attachments/assets/58e4416a-1585-438f-8738-f8a269cbdbfb">